src/python/pants/backend/python/goals/export.py

```diff
@@ -50,6 +50,17 @@ def debug_hint(self) -> str | None:
         return self.resolve
 
 
+@union
+class ExportPythonToolSentinel:
+    """Pythion tools use this as an entry point to say how to export their tool virtualenv."""
+
+
+@dataclass(frozen=True)
+class ExportPythonTool:
```

Comment:
It would be good to subclass `EngineAwareParameter` and implement `debug_hint()` to return `self.resolve_name`